app/controllers/registrations_controller.rb

```diff
@@ -30,11 +30,13 @@ def create
 
   private
 
-  def check_registrations_open_or_valid_invite!
-    return true if AppConfig.settings.enable_registrations? || invite.try(:can_be_used?)
-
-    flash[:error] = params[:invite] ? t("registrations.invalid_invite") : t("registrations.closed")
-    redirect_to new_user_session_path
+  def check_valid_invite!
+    if params[:invite] && !invite.try(:can_be_used?)
```

Comment:
Style/GuardClause: Use a guard clause instead of wrapping the code inside a conditional expression.